### PR TITLE
Migrate to multistage Dockerfile

### PR DESCRIPTION
**Pull request by nipsufn**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FROM node:alpine as builder
+RUN apk update &&\
+	apk add imagemagick &&\
+	apk add git
+RUN npm install mv graceful-fs fs-extra request path walk unzipper child_process ncp
```
Comment (Demonsthere) [inline Dockerfile]: Please use the `package.json` file to download dependencies instead

**Review by Demonsthere — CHANGES_REQUESTED** [review]

**Comment by Demonsthere** [comment]
A few more suggestions:
- please don't delete the `custom-icons` and `class-icons` directories, as they are used by the `update-icons.js` script, and lack of them causes errors in the script:

```
Error [Error: ENOENT: no such file or directory, stat './resources/custom-icons'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'stat',
  path: './resources/custom-icons'
}
```

**Comment by nipsufn** [comment]
I've added back hastly removed directories and contents, improved the Dockerfile.

**Review by Demonsthere — APPROVED** [review]
LGTM :)

**Pull request merged**